mining.go

```diff
@@ -1865,7 +1858,7 @@ mempoolLoop:
 	// Calculate the required difficulty for the block.  The timestamp
 	// is potentially adjusted to ensure it comes after the median time of
 	// the last several blocks per the chain consensus rules.
-	ts, err := medianAdjustedTime(chainState, timeSource)
+	ts, err := medianAdjustedTime(blockManager.chain.BestSnapshot(), timeSource)
```

Comment:
This is in a loop.  Please store a local to the best state before starting the loop and use the local.